src/azure-cli/azure/cli/command_modules/appconfig/feature.py

```diff
@@ -0,0 +1,937 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long
+
+import json
+import time
+import re
+import copy
+
+from knack.log import get_logger
+from knack.util import CLIError
+
+from ._utils import resolve_connection_string, user_confirmation
+from ._azconfig.azconfig_client import AzconfigClient
+from ._azconfig.constants import StatusCodes
+from ._azconfig.exceptions import HTTPException
+from ._azconfig.models import (KeyValue,
+                               ModifyKeyValueOptions,
+                               QueryKeyValueCollectionOptions,
+                               QueryKeyValueOptions)
+from ._featuremodels import (map_keyvalue_to_featureflag,
+                             map_keyvalue_to_featureflagvalue,
+                             FeatureFilter)
+
+
+logger = get_logger(__name__)
+FEATURE_FLAG_PREFIX = ".appconfig.featureflag/"
+FEATURE_FLAG_CONTENT_TYPE = "application/vnd.microsoft.appconfig.ff+json;charset=utf-8"
+
+# Feature commands #
+
+
+def set_feature(cmd,
+                feature,
+                name=None,
+                label=None,
+                description=None,
+                yes=False,
+                connection_string=None):
+    key = FEATURE_FLAG_PREFIX + feature
+
+    # when creating a new Feature flag, these defaults will be used
+    tags = {}
+    default_conditions = {'client_filters': []}
+
+    default_value = {
+        "id": feature,
+        "description": description,
+        "enabled": False,
+        "label": label,
+        "conditions": default_conditions
+    }
+
+    connection_string = resolve_connection_string(cmd, name, connection_string)
+    azconfig_client = AzconfigClient(connection_string)
+    retry_times = 3
+    retry_interval = 1
+    query_options = QueryKeyValueOptions(label=label)
+    for i in range(0, retry_times):
+        retrieved_kv = azconfig_client.get_keyvalue(key, query_options)
+        try:
+            # if kv exists and only content-type is wrong, we can force correct it by updating the kv
+            if retrieved_kv is None:
+                set_kv = KeyValue(
+                    key,
+                    json.dumps(default_value),
+                    label,
+                    tags,
+                    FEATURE_FLAG_CONTENT_TYPE)
+            else:
+                if retrieved_kv.content_type != FEATURE_FLAG_CONTENT_TYPE:
+                    logger.warning(
+                        "This feature contains invalid content-type. If you proceed, the feature flag will be overwritten.")
+                # we make sure that value retrieved is a valid json and only has the fields supported by backend.
```

Comment:
nit: the warning message looks client has a choice to stop?

Reply:
User still has a choice if he hasn't supplied the --yes option. We ask for confirmation after this warning.


---
In reply to: [336140389](https://github.com/Azure/azure-cli/pull/10774#discussion_r336140389) [](ancestors = 336140389)